[TW-1109] Flows restructure (#4996)
postmanlabs/postman-docs: src/pages/docs/postman-flows/concepts/creating-a-form.md

---
title: "Creating a form in Postman Flows"
updated: 2023-08-02
---

> Forms in Postman Flows enable you to seamlessly gather user input and integrate it into your API-driven applications. With Forms, you can create interactive and customizable experiences by collecting data from users through a variety of question types. For example, if you have a flow that gets the weather in a particular city, you could create a form where the user types in the name of the city. The flow runs after the user enters the information and selects **Submit**. Users' input can then be utilized to trigger API requests, perform data manipulation, or generate dynamic outputs. Understanding the key concepts of Forms will help you make the most of this powerful feature in your application development process.

Reviewer: ```suggestion
Forms in Postman Flows enable you to seamlessly gather user input and integrate it into your API-driven applications. With forms, you can create interactive and customizable experiences by collecting data from users through a variety of question types. For example, if you have a flow that gets the weather in a particular city, you could create a form where the user types in the name of the city. The flow runs after the user enters the information and selects **Submit**. Users' input can then be utilized to trigger API requests, perform data manipulation, or generate dynamic outputs. Understanding the key concepts of Forms will help you make the most of this powerful feature in your application development process.
```

Reviewer: forms isn't a branded term (or is it?) so we shouldn't capitalize it. You could do _forms_ in the lede if you're defining it as a new term.

![Forms in Postman Flows](https://assets.postman.com/postman-docs/v10/flows-forms-hero-v10.jpg)

## Contents

* [Form components](#form-components)
* [Form workflow](#form-workflow)

## Form components

The **Form Editor** is where you create and design your forms. It provides an intuitive interface for adding and configuring questions and managing user input.

In Forms, **questions** represent the input fields that users will interact with. Each question can have various question types, such as short answer, paragraph, checkboxes, multiple choice, or dropdown.

* Question Types

>     * Short answer - Creates a single-line text box for users to provide concise responses.
>     * Paragraph - Creates a multiple-line text box for users to enter longer text or descriptions.
>     * Checkboxes - Creates a list of options with checkboxes for users to select multiple choices.
>     * Multiple choice - Creates a list of options with radio buttons for users to select a single choice.
>     * Dropdown - Creates a dropdown list with selectable options for users to choose from.
>     * Required - The **Required** option enables you to make certain questions mandatory, ensuring that users must provide responses before submitting the form.

Reviewer 2: ```suggestion
    * **Short answer** - Creates a single-line text box for users to provide concise responses.
    * **Paragraph** - Creates a multiple-line text box for users to enter longer text or descriptions.
    * **Checkboxes** - Creates a list of options with checkboxes for users to select multiple choices.
    * **Multiple choice** - Creates a list of options with radio buttons for users to select a single choice.
    * **Dropdown** - Creates a dropdown list with options that users can select from.
    * **Required** - Enables you to make certain questions mandatory, ensuring that users must provide responses before submitting the form.
```

---
title: "Create a form in Postman Flows"
updated: 2023-08-02
---

Forms in Postman Flows enable you to seamlessly gather user input and integrate it into your API-driven applications. With forms, you can create interactive and customizable experiences by collecting data from users through a variety of question types. For example, if you have a flow that gets the weather in a particular city, you could create a form where the user types in the name of the city. The flow runs after the user enters the information and selects **Submit**. Users' input can then be utilized to trigger API requests, perform data manipulation, or generate dynamic outputs.

<img alt="Forms in Postman Flows" src="https://assets.postman.com/postman-docs/v10/flows-form-hero-v10.jpg" width="500px">

## Contents

* [Form components](#form-components)

> * [Form workflow](#form-workflow)

Reviewer: I think in general this topic could use structural work. There are really two flows here: how the creator of the form sets it up and puts the link to the form in front of users, and how the end user actually goes the link, uses the form, and the data goes back to flows.

## Form components

> The **Form Editor** enables you to create and design your forms. It provides an intuitive interface for adding and configuring questions and managing user input.

Reviewer: ```suggestion
The _Form Editor_ enables you to create and design your forms. It provides an intuitive interface for adding and configuring questions and managing user input.
```

> In Forms, **questions** represent the input fields that users will interact with. Each question can have various question types, such as short answer, paragraph, checkboxes, multiple choice, or dropdown.

Reviewer: ```suggestion
In Forms, _questions_ represent the input fields that users will interact with. Each question can have various question types, such as short answer, paragraph, checkboxes, multiple choice, or dropdown.
```

> * Question Types

Reviewer: Change the first bullet to an intro sentence and then outdent the rest of these by one. (No sense in having a first-level list with just one item)

    * **Checkboxes** - Creates a list of options with checkboxes for users to select multiple choices.
    * **Multiple choice** - Creates a list of options with radio buttons for users to select a single choice.
    * **Dropdown** - Creates a dropdown list with options that users can select from.
    * **Required** - Enables you to make certain questions mandatory, ensuring that users must provide responses before submitting the form.

## Form workflow

To create a form, select a flow, then select **Form** from the bottom of the **Flows** sidebar.

<img alt="Create a form" src="https://assets.postman.com/postman-docs/v10/flows-create-form-v10.jpg" width="500px">

You can **name your form, add a description, and add questions** by selecting the **more actions icon <img alt="More actions icon" src="https://assets.postman.com/postman-docs/icon-more-actions-v9.jpg#icon" width="16px"> > Edit** and using the available question types. Customize each question to gather specific data from users.

<img src="https://assets.postman.com/postman-docs/v10/flows-edit-form-v10.gif" alt="Edit a form" fetchpriority="low" loading="lazy" >

> You can **rename, duplicate, or delete questions** as needed. You also have the flexibility to reorder questions to enhance the user experience.

Reviewer: There's a lot of bold in this that shouldn't be bold.
